src/wp-includes/class-wp-script-modules.php

```diff
@@ -311,31 +331,71 @@ private function get_highest_fetchpriority( array $ids ): string {
 	}
 
 	/**
-	 * Prints the enqueued script modules using script tags with type="module"
-	 * attributes.
+	 * Prints the enqueued script modules in head.
+	 *
+	 * @since 6.9.0
+	 */
+	public function print_head_enqueued_script_modules() {
+		$script_modules            = $this->get_marked_for_enqueue();
+		$sorted_script_modules_ids = $this->get_sorted_dependencies( array_keys( $script_modules ) );
+		foreach ( $sorted_script_modules_ids as $id ) {
+			if ( isset( $script_modules[ $id ] ) && ! $script_modules[ $id ]['in_footer'] ) {
+				// If any dependency is set to be printed in footer, skip printing this module in head.
+				$dependencies = $this->get_dependencies( array( $id ) );
+				foreach ( $dependencies as $dependency_id => $dependency ) {
+					if ( in_array( $dependency_id, $this->queue, true ) && $dependency['in_footer'] ) {
+						continue 2;
+					}
+				}
+				$this->done[] = $id;
+				$this->print_script_module( $id, $script_modules[ $id ] );
+			}
+		}
+	}
+
+	/**
+	 * Prints the enqueued script modules in footer.
 	 *
 	 * @since 6.5.0
 	 */
 	public function print_enqueued_script_modules() {
-		foreach ( $this->get_marked_for_enqueue() as $id => $script_module ) {
-			$args = array(
-				'type' => 'module',
-				'src'  => $this->get_src( $id ),
-				'id'   => $id . '-js-module',
-			);
-
-			$dependents    = $this->get_recursive_dependents( $id );
-			$fetchpriority = $this->get_highest_fetchpriority( array_merge( array( $id ), $dependents ) );
-			if ( 'auto' !== $fetchpriority ) {
-				$args['fetchpriority'] = $fetchpriority;
-			}
-			if ( $fetchpriority !== $script_module['fetchpriority'] ) {
-				$args['data-wp-fetchpriority'] = $script_module['fetchpriority'];
+		$script_modules            = $this->get_marked_for_enqueue();
+		$sorted_script_modules_ids = $this->get_sorted_dependencies( array_keys( $script_modules ) );
+		foreach ( $sorted_script_modules_ids as $id ) {
+			if ( isset( $script_modules[ $id ] ) ) {
+				$this->done[] = $id;
+				$this->print_script_module( $id, $script_modules[ $id ] );
```

Comment:
With my other suggestion below, this can be simplified:

```suggestion
		foreach ( $this->get_sorted_dependencies( $this->queue ) as $id ) {
			$this->print_script_module( $id );
```